apps/arweave/src/ar_mining_server.erl

```diff
@@ -576,18 +590,25 @@ prepare_solution(poa2, Candidate,
 		partition_upper_bound = PartitionUpperBound } = Candidate,
 	case read_poa(RecallByte2, Chunk2, MiningAddress) of
 		error ->
-			{_RecallRange1Start, RecallRange2Start} = ar_block:get_recall_range(H0,
-					PartitionNumber, PartitionUpperBound),
-			?LOG_WARNING([{event, mined_block_but_failed_to_read_chunk_proofs},
-					{recall_byte2, RecallByte2},
-					{recall_range_start2, RecallRange2Start},
-					{nonce, Nonce},
-					{partition, PartitionNumber},
-					{mining_address, ar_util:safe_encode(MiningAddress)}]),
-			ar:console("WARNING: we have mined a block but failed to fetch "
-					"the chunk proofs required for publishing it. "
-					"Check logs for more details~n"),
-			error;
+			ar:console("WARNING: we have mined a block but did not find the PoA2 proofs "
+					"locally - searching the peers...~n"),
+			case fetch_poa_from_peers(RecallByte2) of
```

Comment:
This part of ar_mining_worker would also benefit from this change, right? https://github.com/ArweaveTeam/arweave/blob/master/apps/arweave/src/ar_mining_worker.erl#L437

I think that bit is invoked in coordinated mining